modules/cognito/main.tf

```diff
@@ -0,0 +1,48 @@
+resource aws_cognito_user_pool this {
+  name = var.cluster_name
+}
+
+resource aws_cognito_user_pool_domain this {
+  domain          = "auth.${var.domain}"
+  certificate_arn = module.cognito_acm.this_acm_certificate_arn
+  user_pool_id    = aws_cognito_user_pool.this.id
+}
+
+resource aws_route53_record this {
+  name    = aws_cognito_user_pool_domain.this.domain
+  type    = "A"
+  zone_id = var.zone_id
+  alias {
+    evaluate_target_health = false
+    name                   = aws_cognito_user_pool_domain.this.cloudfront_distribution_arn
+    zone_id                = "Z2FDTNDATAQYW2"
```

Comment:
hardcode?

Reply:
Z2FDTNDATAQYW2 is always the hosted zone ID when you create an alias record that routes traffic to a CloudFront distribution.